src/main/scala/com/databricks/spark/csv/package.scala

```diff
@@ -43,21 +43,23 @@ package object csv {
   }
 
   implicit class CsvSchemaRDD(dataFrame: DataFrame) {
-    def saveAsCsvFile(path: String): Unit = {
+    def saveAsCsvFile(path: String, parameters: Map[String, String] = Map()): Unit = {
       // TODO(hossein): For nested types, we may want to perform special work
-      val header = dataFrame.columns.map(c => s""""$c"""").mkString(",")
+      val delimiter = parameters.getOrElse("delimiter", ",")
+      val generateHeader = parameters.getOrElse("generateHeader", "false").toBoolean
+      val header = dataFrame.columns.map(c => s""""$c"""").mkString(delimiter)
       val strRDD = dataFrame.rdd.mapPartitions { iter =>
         new Iterator[String] {
-          var firstRow: Boolean = true
+          var firstRow: Boolean = generateHeader
 
           override def hasNext = iter.hasNext
 
           override def next: String = {
             if (firstRow) {
               firstRow = false
-              header + "\n" + iter.next.mkString(",")
+              header + "\n" + iter.next.mkString(delimiter)
```

Comment:
Note that I had to change the logic slightly (so that "INSERT OVERWRITE WORKS"). In particular, I have to use the same delimiter as before.
Also, the default logic was to always add a header row, while it might be the case that we already have them. Thus the default behavior should be to write the RDD as is, and if the user who is using this functions wants us to add a header, we will pass `generateHeader ` in parameters (I'll add this to README).